pandora-common/src/character/characterData.ts

```diff
@@ -21,13 +22,41 @@ export const CHARACTER_DEFAULT_PUBLIC_SETTINGS: Readonly<ICharacterPublicSetting
 	pronoun: 'she',
 };
 
+export const AttributePreferenceTypeSchema = z.enum(['normal', 'prevent', 'doNotRender']);
+export type AttributePreferenceType = z.infer<typeof AttributePreferenceTypeSchema>;
+
+export const AttributePreferenceSchema = z.object({
+	base: AttributePreferenceTypeSchema.default('normal'),
+});
+export type AttributePreference = z.infer<typeof AttributePreferenceSchema>;
+
+export const AssetPreferenceTypeSchema = z.enum(['favorite', 'normal', 'maybe', 'prevent', 'doNotRender']);
+export type AssetPreferenceType = z.infer<typeof AssetPreferenceTypeSchema>;
+
+export const AssetPreferenceSchema = z.object({
+	base: AssetPreferenceTypeSchema.default('normal'),
+});
+export type AssetPreference = z.infer<typeof AssetPreferenceSchema>;
+
+export const AssetPreferencesSchema = z.object({
+	attributes: z.record(z.string(), AttributePreferenceSchema).default({}),
+	assets: z.record(AssetIdSchema, AssetPreferenceSchema).default({}),
+});
+export type AssetPreferences = z.infer<typeof AssetPreferencesSchema>;
+
+export const ASSET_PREFERENCES_DEFAULT: Readonly<AssetPreferences> = Object.freeze<AssetPreferences>({
+	attributes: {},
+	assets: {},
+});
+
 /** Data about character, that is visible to everyone in same room */
 export const CharacterPublicDataSchema = z.object({
 	id: CharacterIdSchema,
 	accountId: AccountIdSchema,
 	name: CharacterNameSchema,
 	profileDescription: z.string().default(''),
 	settings: CharacterPublicSettingsSchema.default(CHARACTER_DEFAULT_PUBLIC_SETTINGS),
+	preferences: AssetPreferencesSchema.default(ASSET_PREFERENCES_DEFAULT),
```

Comment:
Naming it just `preferences` seems weird when it specifically only concerns asset preferences - maybe name it that - `assetPreferences`?